### PR TITLE
xc: init at 0.0.154

### PR DESCRIPTION
**Pull request by joerdav**
###### Description of changes

xc is a makefile alternative that allows you to defined tasks to be run in markdown.
"Create executable documentation"
xcfile.dev/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Zaechus — CHANGES_REQUESTED** [review]
##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [ ] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [ ] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden

##### Possible improvements
`description` should not end with a period. It should also be changed to something else like "Markdown defined task runner".

##### Comments
`xc` binary works as intended.

**Comment by joerdav** [comment]
Thankyou for the feedback @Zaechus I've updated that to fit the guidelines.

**Review by Zaechus — APPROVED** [review]
Thank you, looks good.

**Comment by joerdav** [comment]
@Zaechus Hey, from your experience how long does it talke for a PR like this to get merged?

File: pkgs/development/tools/xc/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "joerdav";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-GJBSPO0PffGdGAHofd1crEFXJi2xqgd8Vk2/g4ff+E4=";
+  };
+
+  vendorHash = "sha256-XDJdCh6P8ScSvxY55ExKgkgFQqmBaM9fMAjAioEQ0+s=";
+
+  doCheck = false;
```
Comment (K900) [inline pkgs/development/tools/xc/default.nix:16]: Is there a reason for this? Maybe add a comment here.

File: pkgs/development/tools/xc/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  nativeBuildInputs = [ installShellFiles ];
```
Comment (K900) [inline pkgs/development/tools/xc/default.nix:18]: Seems unused?

File: pkgs/development/tools/xc/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  subPackages = [ "cmd/xc" ];
+
+  ldflags = [
```
Comment (K900) [inline pkgs/development/tools/xc/default.nix:22]: Any specific reason to deviate from the defaults here?

**Review by K900 — CHANGES_REQUESTED** [review]
Also, please squash the review changes into the init commit.

File: pkgs/development/tools/xc/default.nix (hunk 1 above)
Comment (joerdav) [reply pkgs/development/tools/xc/default.nix:16]: I used another package as a starting point, should have omitted it, will address!

**Comment by joerdav** [comment]
Thanks @K900 I've addressed those changes.

File: pkgs/development/tools/xc/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
```
Comment (K900) [inline pkgs/development/tools/xc/default.nix:1]: Don't need `installShellFiles` here either.

File: pkgs/development/tools/xc/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner = "joerdav";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-GJBSPO0PffGdGAHofd1crEFXJi2xqgd8Vk2/g4ff+E4=";
+  };
+
+  vendorHash = "sha256-XDJdCh6P8ScSvxY55ExKgkgFQqmBaM9fMAjAioEQ0+s=";
+
+  subPackages = [ "cmd/xc" ];
```
Comment (K900) [inline pkgs/development/tools/xc/default.nix:16]: This also doesn't seem to be necessary, but I'm not familiar enough with Go stuff to really tell, so please disregard me if I'm wrong here.

File: pkgs/development/tools/xc/default.nix (hunk 4 above)
Comment (joerdav) [reply pkgs/development/tools/xc/default.nix]: Good point there, thanks.

File: pkgs/development/tools/xc/default.nix (hunk 5 above)
Comment (joerdav) [reply pkgs/development/tools/xc/default.nix]: No I think your right, I only have one binary in there so I don't have to specify.

**Pull request merged**